client/src/main/java/com/alibaba/nacos/client/naming/NacosNamingService.java

```diff
@@ -156,6 +159,45 @@ public void batchRegisterInstance(String serviceName, String groupName, List<Ins
         clientProxy.batchRegisterService(serviceName, groupName, instances);
     }
     
+    @Override
+    public void batchDeRegisterInstance(String serviceName, String groupName, List<Instance> instances)
+            throws NacosException {
+        NamingUtils.batchCheckInstanceIsLegal(instances);
+        // Filter the instances that need to be de registered
+        instances = getRetainInstance(serviceName, groupName, instances);
+        clientProxy.batchRegisterService(serviceName, groupName, instances);
+    }
+    
+    /**
+     * Get instance list that need to be deregistered.
+     * @param serviceName   service name
+     * @param groupName    group name
+     * @param instances   instance list
+     * @return  instance list that need to be deregistered.
+     */
+    private List<Instance> getRetainInstance(String serviceName, String groupName, List<Instance> instances) throws NacosException {
+        if (CollectionUtils.isEmpty(instances)) {
+            throw new NacosException(NacosException.INVALID_PARAM,
+                    String.format("[Batch deRegistration] need deRegister instance is empty, instances: %s,", instances));
+        }
+        
+        instances.forEach(instance -> instance.setServiceName(NamingUtils.getGroupedName(serviceName, groupName)));
+        List<Instance> allInstances = getAllInstances(serviceName, groupName);
```

Comment:
GetAllInstances？ this method will get all instance which include other client register.
I think should use cachedInstance in this client.

And the getRetain logic I think should be move into GrpcClientProxy.

Reply:
I've solved it